h/services/annotation_stats.py

```diff
@@ -16,36 +14,45 @@ class AnnotationStatsService(object):
     def __init__(self, request):
         self.request = request
 
-    def user_annotation_counts(self, userid):
-        """Return the count of annotations for this user."""
+    def user_annotation_count(self, userid):
+        """
+        Return the count of searchable top level annotations for this user.
+
+        If the logged in user has this userid, private annotations will be
+        included in this count, otherwise they will not.
+        """
+        params = MultiDict({'limit': 0, 'user': userid})
+        return self._search(params)
 
-        annotations = self.request.db.query(Annotation). \
-            filter_by(userid=userid, deleted=False). \
-            options(sa.orm.load_only('groupid', 'shared')).subquery()
-        grouping = sa.case([
-            (sa.not_(annotations.c.shared), 'private'),
-            (annotations.c.groupid == '__world__', 'public'),
-        ], else_='group')
+    def total_user_annotation_count(self, userid):
+        """
+        Return the count of all annotations for this user.
 
-        result = dict(self.request.db.query(grouping, sa.func.count(annotations.c.id)).group_by(grouping).all())
-        for key in ['public', 'group', 'private']:
-            result.setdefault(key, 0)
+        This disregards permissions, private/public, etc and returns the
+        total number of annotations the user has made (including replies).
+        """
+        params = MultiDict({'limit': 0, 'user': userid})
 
-        result['total'] = result['public'] + \
-            result['group'] + \
-            result['private']
+        search = Search(self.request, stats=self.request.stats)
+        search.clear()
+        search.append_modifier(Limiter())
+        search.append_modifier(DeletedFilter())
+        search.append_modifier(UserFilter())
 
-        return result
+        search_result = search.run(params)
+        return search_result.total
 
     def group_annotation_count(self, pubid):
         """
         Return the count of searchable top level annotations for this group.
         """
+        params = MultiDict({'limit': 0, 'group': pubid})
```

Comment:
It looks like we're always passing `"limit": 0` when calling `_search`, so this parameter could be "pushed down" into that method.